libs/features/mentor-courses/src/lib/features-mentor-courses.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect, useState } from 'react';
+import { SharedCourseCard } from '@ppm/shared/course-card';
+import { useSelector, useDispatch } from 'react-redux';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import { createStructuredSelector } from 'reselect';
+import { coursesActions, coursesSelectors } from '@ppm/data-access/courses';
+import { likesActions } from '@ppm/data-access/likes';
+import {
+  userProfileActions,
+  userProfileSelectors,
+} from '@ppm/data-access/user-profile';
+import { SharedCourseDetails } from '@ppm/shared/course-details';
+import { SharedCourseInfo } from '@ppm/shared/course-info';
+import { mentorsActions, mentorsSelectors } from '@ppm/data-access/mentors';
+// import { likesActions, likesSelectors } from '@ppm/data-access/likes';
+import './features-mentor-courses.scss';
+import { LikeEnum } from 'libs/data-access/likes/src/lib/types';
+import { RouteComponentProps, RouteProps } from 'react-router-dom';
+
+interface RouteInfo extends RouteProps {
+  params: {
+    id: string;
+  };
+  path?: string | string[];
+}
+
+const stateSelector = createStructuredSelector({
+  courses: coursesSelectors.selectCourses(),
+  loading: coursesSelectors.selectLoading(),
+  profile: userProfileSelectors.selectUserProfile(),
+  loadedProfile: userProfileSelectors.selectOtherProfile(),
+});
+
+export const FeaturesMentorCourses = (props: {
+  history: History;
+  match: RouteComponentProps<RouteInfo>;
+}) => {
+  const dispatch = useDispatch();
+  const { courses, loading } = useSelector(stateSelector);
```

Comment:
you can use it in single hook like this:   const { courses, loading, loadedProfile, profile, mentors } = useSelector(stateSelector);
Don't do duplications.

Reply:
Makes a lot of sense! Solved